packages/wallet/src/SingleAddressWallet/prepareTx.ts

```diff
@@ -1,84 +0,0 @@
-import { Cardano } from '@cardano-sdk/core';
-import { FinalizeTxProps, InitializeTxProps, ObservableWallet } from '../types';
-import { Logger } from 'ts-log';
-import { Observable, combineLatest, filter, firstValueFrom, lastValueFrom, map, take } from 'rxjs';
-import { createTransactionInternals, ensureValidityInterval } from '../Transaction';
-import { defaultSelectionConstraints } from '@cardano-sdk/tx-construction';
-
-export interface PrepareTxDependencies {
-  wallet: Pick<ObservableWallet, 'tip$' | 'protocolParameters$' | 'addresses$' | 'genesisParameters$'> & {
-    delegation: Pick<ObservableWallet['delegation'], 'rewardAccounts$'>;
-    utxo: Pick<ObservableWallet['utxo'], 'available$'>;
-    syncStatus: Pick<ObservableWallet['syncStatus'], 'isSettled$'>;
-  };
-  signer: {
-    stubFinalizeTx(props: FinalizeTxProps): Observable<Cardano.Tx>;
-  };
-  logger: Logger;
-}
-
-export const createTxPreparer =
-  ({ wallet, signer, logger }: PrepareTxDependencies) =>
-  async (props: InitializeTxProps) => {
-    await firstValueFrom(wallet.syncStatus.isSettled$.pipe(filter((isSettled) => isSettled)));
```

Comment:
I removed waiting for `isSettled` here because `GenericTxBuilder` is not related to any wallet, but to providers, so there is no point in having a `isSettled` methods. Callers should make sure that providers do resolve unless they are settled.
A possible way to reintroduce it is in `SingleAddresWallet.getTxBuilderDependencies`, as a precondition for any provider value.

Reply:
https://github.com/input-output-hk/cardano-js-sdk/pull/712/commits/ba1c181671ef0b27298f07e4a593f3e4a16a5371